workers/unity/Assets/Playground/Scripts/Player/ConnectButtonClickSystem.cs

```diff
@@ -0,0 +1,93 @@
+using Improbable.Gdk.Core;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+using UnityEngine.UI;
+#if UNITY_ANDROID
+using Improbable.Gdk.Android;
+
+#endif
+
+namespace Playground
+{
+    [UpdateBefore(typeof(ConnectClientSystem))]
+    internal class ConnectButtonClickSystem : ComponentSystem
+    {
+        private struct Data
+        {
+            public readonly int Length;
+            [ReadOnly] public EntityArray Entity;
+            [ReadOnly] public ComponentDataArray<WorkerEntityTag> DenotesWorker;
+        }
+
+        private InputField connectParamInputField;
+        private Button connectButton;
+        private bool clicked;
+        private WorkerBase worker;
+        private Text errorField;
+
+        [Inject] private Data data;
+
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+            worker = WorkerRegistry.GetWorkerForWorld(World);
+            connectParamInputField = GameObject.Find("ConnectParam").GetComponent<InputField>();
+            connectButton = GameObject.Find("ConnectButton").GetComponent<Button>();
+            connectButton.onClick.AddListener(IsClicked);
+            errorField = GameObject.Find("ConnectionError").GetComponent<Text>();
+            if (!Application.isMobilePlatform)
+            {
+                connectParamInputField.gameObject.SetActive(false);
+            }
+        }
+
+        protected override void OnUpdate()
+        {
+            if (clicked)
+            {
+                for (var i = 0; i < data.Length; i++)
+                {
+                    PostUpdateCommands.AddComponent(data.Entity[i], new ConnectButtonClicked());
+                    connectButton.gameObject.SetActive(false);
+                    errorField.text = "";
+                    clicked = false;
+                }
+            }
+        }
+
+        private void IsClicked()
+        {
+            clicked = true;
+#if UNITY_ANDROID
+            if (Application.isMobilePlatform)
+            {
+                if (DeviceInfo.IsAndroidStudioEmulator() && connectParamInputField.text.Equals(""))
+                {
+                    worker.ConnectionConfig = ReceptionistConfig.CreateConnectionConfigForAndroidEmulator();
+                }
+                else
+                {
+                    SetConnectionParameters(connectParamInputField.text);
+                }
+            }
+#endif
+        }
+
+
+        private void SetConnectionParameters(string param)
+        {
+            if (IsIpAddress(param))
+            {
+                worker.ConnectionConfig = ReceptionistConfig.CreateConnectionConfigForPhysicalAndroid(param);
+            }
+
+            // TODO: UTY-558 else -> cloud connection
+        }
+
+        private static bool IsIpAddress(string param)
+        {
+            return param.Split('.').Length == 4;
```

Comment:
use regex to check if it's an ip address: https://stackoverflow.com/questions/5284147/validating-ipv4-addresses-with-regexp

Reply:
I was thinking about doing this but it is generally advised ( https://www.perlmonks.org/index.pl?node_id=221512 ) not do to that. Maybe in our case it doesn't matter because we can assume they will be only entering it in human-readable format?

Reply:
I think it's fine, because with the current check you are also checking for 4 dots, which the page also pointed out as not being correct

Reply:
the other option would be to use: https://msdn.microsoft.com/en-us/library/system.net.ipaddress.tryparse(v=vs.110).aspx

though that seems like a bit of an overhead right now